gateway/sensor.go

```diff
@@ -67,75 +45,17 @@ func init() {
 
 // Validate ensures all parts of the config are valid.
 func (conf *Config) Validate(path string) ([]string, error) {
-	for _, d := range conf.Devices {
-		if d.DecoderPath == "" {
-			return nil, resource.NewConfigValidationError(path,
-				errors.New("decoder path is required"))
-		}
-		switch d.JoinType {
-		case "ABP":
-			return d.validateABPAttributes(path)
-		case "OTAA", "":
-			return d.validateOTAAAttributes(path)
-		default:
-			return nil, resource.NewConfigValidationError(path,
-				errors.New("join type is OTAA or ABP - Default is OTAA"))
-		}
-
-	}
-	return nil, nil
-}
-
-func (conf *DeviceConfig) validateOTAAAttributes(path string) ([]string, error) {
-	if conf.DevEUI == "" {
-		return nil, resource.NewConfigValidationError(path,
-			errors.New("dev EUI is required for OTAA join type"))
-	}
-	if len(conf.DevEUI) != 16 {
+	if conf.ResetPin == nil {
 		return nil, resource.NewConfigValidationError(path,
-			errors.New("dev EUI must be 8 bytes"))
+			errors.New("reset pin is required"))
 	}
-	if conf.AppKey == "" {
+	if conf.Bus != 0 && conf.Bus != 1 {
 		return nil, resource.NewConfigValidationError(path,
-			errors.New("app key is required for OTAA join type"))
-	}
-	if len(conf.AppKey) != 32 {
-		return nil, resource.NewConfigValidationError(path,
-			errors.New("app key must be 16 bytes"))
+			errors.New("spi bus can be 0 or 1 - default 0"))
```

Comment:
in the constructor or reconfigure, is the spi bus set to 0 if it's not 0 or 1? or is this just to inform that you can set it to 0 if you aren't sure what it should be?